RELEASING/Dockerfile.from_local_tarball

```diff
@@ -56,7 +56,6 @@ RUN npm ci \
 
 WORKDIR /home/superset/apache-superset-incubating-$VERSION
 RUN pip install --upgrade setuptools pip \
-    && pip install -r requirements.txt \
```

Comment:
That makes the docker image non deterministic which I think is an issue. Personally I vote for having a very deterministic docker build.

Reply:
Valid point. Perhaps default to using `requirements.txt` but add an option `pip` or similar to the script (similar to how `local` works now) to skip this step and pull in whatever a vanilla `pip install` would pull in?

Reply:
Can we assume that people would just add docker layers that would force install deps on top or our base image? Maybe I'm missing something but I think it's a better option than parameterizing the Dockerfile.

Reply:
I think any serious user will do that. However, as long as local instances of Superset are being installed via `pip` without using a prebuilt image, getting the recommended pinned deps will be difficult without doing a wget to the github branch (to my knowledge `pip` does not currently support proper locking). But I haven't looked into this in detail lately, so there may better ways to solve this problem.

Reply:
We could be more restrictive on the `setup.py` ranges or even pinning deps there, but that makes it hard for people to override in their local environments. But since Superset is not a library but a self-standing app, it's more acceptable to pin deps.

An alternative to catch range issues would be to setup a nightly build that runs a `pip install .` (without using pinned deps in requirements.txt). When that nightly build fails we'd look into the issue and adjust ranges...

Reply:
Not a simple issue. I'm all for pinned dependencies, but @villebro has a point here, since users will be using `pip install apache-superset`. Yet a good deployment should always include a requirements.txt to pin the dependencies. This docker serves has a quick test to catch immediate issues with the release prior to pushing to Apache.

So I think it makes sense not to pin here, it's a plus for us to early catch dependency problems. A nightly build to test possible problems is a plus also.

Finally being too restrictive on `setup.py` can trigger hard to solve issues, for example not allowing user's to bump dependencies that have problems on specific platforms, etc

Reply:
One important question is whether a docker image built from certain SHA in the repo be 
"as deterministic as can be" or not.

I hate the idea that the same Apache tarball-based docker build would potentially be different, succeed or failed depending on when it is built.

Personally I vote for using the lock files (requirements.txt) for the purpose of building dockers. Make the docker as builds as deterministic as can be.

We do need another mechanism to catch issues around moving builds when doing `pip install apache-superset`. For that I suggest a nightly where we do not use the lock files.

Reply:
@craig-rueda thoughts?

Reply:
Still feel the same in this case, even after the discussion. 😆 For releasing, I think it makes sense to use setup.py, whereas for building docker images that are shipped to docker hub, we should use pinned versions. I think dynamic versions in this case give a better feel for what the end user will see when running 'pip install superset'

Reply:
Though here we're not trying to simulate `pip install apache-superset`, the goal is to provide build instructions or a reference dockerfile for building from the signed, official tarball. Ideally everyone building any particular version has the same recipe and gets to the same docker (unless they actively want to alter/deviate).

Side note: at the time where I wrote it there was no official dockerfile in the base of the repo, now that there's one we may be able to just use that one instead (?)